plugins/jobtype/src/azkaban/jobtype/HadoopJobUtils.java

```diff
@@ -0,0 +1,255 @@
+   *   1. ./lib/abc
+   *   2. ./lib/abc.jar
+   * 
+   * This method will use prefix matching to find any jar that is the form of abc*.jar,
+   * so that users can bump jar versions without doing modifications to their Hadoop DSL.
+   * 
+   * This method will throw an Exception if more than one jar that matches the prefix is found
+   * 
+   * @param workingDirectory
+   * @param userSpecifiedJarName
+   * @return the resolved actual jar file name to execute
+   */
+  public static String resolveExecutionJarName(String workingDirectory,
+          String userSpecifiedJarName, Logger log) {
+
+    log.debug("Resolving execution jar name: working directory: " + workingDirectory
+            + ", user specified name: " + userSpecifiedJarName);
+
+    // in case user decides to specify with abc.jar, instead of only abc
+    if (userSpecifiedJarName.endsWith(".jar"))
+      userSpecifiedJarName = userSpecifiedJarName.replace(".jar", "");
+
+    // can't use java 1.7 stuff, reverting to a slightly ugly implementation
+    String userSpecifiedJarPath = String.format("%s/%s", workingDirectory, userSpecifiedJarName);
+    int lastIndexOfSlash = userSpecifiedJarPath.lastIndexOf("/");
+    final String jarname = userSpecifiedJarPath.substring(lastIndexOfSlash + 1);
+    final String dirName = userSpecifiedJarPath.substring(0, lastIndexOfSlash);
+    log.debug("Resolving execution jar name: dirname: " + dirName + ", jar name:  " + jarname);
+    File[] potentialExecutionJarList = new File(dirName).listFiles(new FilenameFilter() {
+      @Override
+      public boolean accept(File dir, String name) {
+        if (name.startsWith(jarname.toString()))
+          return true;
+        else
+          return false;
+      }
+    });
+
+    if (potentialExecutionJarList.length == 0) {
+      throw new IllegalStateException("unable to find execution jar for Spark at path: "
+              + userSpecifiedJarPath);
+    } else if (potentialExecutionJarList.length > 1) {
+      throw new IllegalStateException(
+              "I find more than one matching instance of the execution jar at the path, don't know which one to use: "
+                      + userSpecifiedJarPath);
+    }
+
+    String resolvedJarName = potentialExecutionJarList[0].toString();
+    log.debug("Resolving execution jar name: resolvedJarName: " + resolvedJarName);
+    return resolvedJarName;
+  }
+
+  /**
+   * <pre>
+   * Uses YarnClient to kill the job on HDFS.
+   * Using JobClient only works partially:
+   *   If yarn container has started but spark job haven't, it will kill
+   *   If spark job has started, the cancel will hang until the spark job is complete
+   *   If the spark job is complete, it will return immediately, with a job not found on job tracker
+   * </pre>
+   * 
+   * @param applicationId
+   */
+  public static void killJobOnCluster(String applicationId, Logger log) {
+    try {
+      YarnConfiguration yarnConf = new YarnConfiguration();
```

Comment:
Note: we should update Azkaban document to mention that these changes will not work with hadoop-1 ? yarn stuff